### ISSUE TITLE
feature request: ignore specific lines instead of entire file

### ISSUE THREAD

**Issue by Barabazs** [google/yamlfmt#126]
I would like to mark a line as to be ignored by yamlfmt. Similar to the use of `# noqa` and similar ignore syntaxes.

I'm using yamlfmt to format CircleCI config files, which references variables as `<< foo.bar >>`  and this should not be reformatted. But when limiting line length to N, yamlfmt will try to break it down in smaller lines.

**Comment by braydonk** [comment]
Hi @Barabazs thanks for the feature request. This is a pretty reasonable request, however it would be really hard to implement given the current architecture of yamlfmt.

Since it (along with some other requested features) likely requires writing a custom yaml parser, it is probably too big of an undertaking for me to try and address it soon. I would like to be able to make it happen some day though. Sorry :disappointed:

**Comment by Barabazs** [comment]
Hi @braydonk thank you for  the prompt response. I totally understand and respect your position.
I never programmed in Go, otherwise I might give it a try with some guidance.

**Comment by braydonk** [comment]
Yeah I understand. Even though imo Go is quite an easy language to pick up, this task would probably be quite challenging.

If anything changes and I'm able to spend more than just spare time on this project, then this would likely be the first major thing I take on. So never say never!
